tests/test-mongoc-retryable-writes.c

```diff
@@ -625,14 +626,11 @@ test_retryable_writes_cb (bson_t *scenario)
       test_framework_set_ssl_opts (client);
       mongoc_uri_destroy (uri);
 
-      /* TODO: Remove this once start_session initializes the topology */
-      {
-         bool success;
-
-         success = mongoc_client_command_simple (
-            client, "admin", tmp_bson ("{'ping': 1}"), NULL, NULL, &error);
-         ASSERT_OR_PRINT (success, error);
-      }
+      /* clean up in case a previous test aborted */
+      server_id = mongoc_topology_select_server_id (
+         client->topology, MONGOC_SS_WRITE, NULL, &error);
+      ASSERT_OR_PRINT (server_id, error);
+      deactivate_fail_point (client, server_id);
```

Comment:
Just saw this PR, but I ended up implementing a similar solution locally. This does address failures in subsequent tests within the same suite, but I don't think it will help if the last test in this suite leaves the fail point activated and control moves on to other test groups.

Is there any way to have the test suite itself execute a callback after a failed test? When running in the default mode, which forks per test, are we able to have the parent process execute a cleanup function? If using non-fork mode (with `-f`), the main process exits on the first failure, so I assume we have no control there.

Reply:
Yeah, alas there's no post-test cleanup hook in our suite. That'd be a nice feature.